tests/test_file_writer.py

```diff
@@ -22,3 +41,141 @@ def test_MantarrayH5FileCreator__sets_file_name_and_userblock_size_and_file_vers
+
+
+def test_h5_file_trimmer__When_start_arg_is_negative__Then_raises_an_error():
+    with pytest.raises(ValidationCollectionMinimumValueError):
+        h5_file_trimmer(PATH_TO_GENERIC_0_4_2_FILE, from_start=-10, from_end=0)
+
+
+def test_h5_file_trimmer__When_start_arg_is_not_an_int__Then_raises_an_error():
+    with pytest.raises(ValidationCollectionNotAnIntegerError):
+        h5_file_trimmer(PATH_TO_GENERIC_0_4_2_FILE, from_start=1.7, from_end=0)
+
+
+def test_h5_file_trimmer__When_end_arg_is_not_valid__Then_raises_an_error():
+    with pytest.raises(ValidationCollectionMinimumValueError):
+        h5_file_trimmer(PATH_TO_GENERIC_0_4_2_FILE, from_start=0, from_end=-1)
+
+
+def test_h5_file_trimmer__When_both_args_are_None__Then_raises_an_error():
+    with pytest.raises(UnsupportedArgumentError):
+        h5_file_trimmer(PATH_TO_GENERIC_0_4_2_FILE, from_start=None, from_end=None)
+
+
+def test_h5_file_trimmer__When_both_file_path_isnt_supported__Then_raises_an_error():
+    EXPECTED_PATH_D6 = os.path.join(
+        PATH_OF_CURRENT_FILE,
+        "2020_08_04_build_775",
+        "MA20001010__2020_08_04_220041__D6.h5",
+    )
+    with pytest.raises(UnsupportedMantarrayFileVersionForTrimmingError):
+        h5_file_trimmer(EXPECTED_PATH_D6, from_start=10, from_end=10)
+
+
+def test_h5_file_trimmer__When_invoked_on_a_file__Then_the_new_file_has_old_metadata_except_for_the_three_metadata_pertaining_to_trimming():
+    new_file_path = h5_file_trimmer(PATH_TO_GENERIC_0_4_2_FILE, 200, 200)
+
+    wf = WellFile(new_file_path)
+    old_wf = WellFile(PATH_TO_GENERIC_0_4_2_FILE)
+
+    # old metadata (since it is all copied by default, testing a subset seems reasonable for now)
+    assert wf.get_h5_attribute(str(WELL_INDEX_UUID)) == old_wf.get_h5_attribute(
+        str(WELL_INDEX_UUID)
+    )
+    assert wf.get_h5_attribute(str(WELL_NAME_UUID)) == old_wf.get_h5_attribute(
+        str(WELL_NAME_UUID)
+    )
+
+    # new metadata
+    assert not wf.get_h5_attribute(
+        str(IS_FILE_ORIGINAL_UNTRIMMED_UUID)
+    )  # Anna (1/20/21): tried using `is False` but got weird errors saying `assert False is False` failed...unsure why
+    assert wf.get_h5_attribute(str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID)) == 160
+    assert wf.get_h5_attribute(str(TRIMMED_TIME_FROM_ORIGINAL_END_UUID)) == 160
+
+    wf.get_h5_file().close()  # safe clean-up when running CI on windows systems
+    old_wf.get_h5_file().close()  # safe clean-up when running CI on windows systems
+
+
+def test_h5_file_trimmer__When_invoked_on_a_file_with_too_much_time_trimmed__Then_raises_TooTrimmedError():
+    with pytest.raises(TooTrimmedError):
+        h5_file_trimmer(
+            PATH_TO_GENERIC_0_4_2_FILE, from_start=6000000, from_end=2000000
+        )
+
+
+def test_h5_file_trimmer__When_invoked_on_a_0_4_1_file_with_args_in_before_time_points__Then_the_new_file_has_trimmed_raw_referene_and_tissue_data():
+    new_file_path = h5_file_trimmer(PATH_TO_GENERIC_0_4_2_FILE, 70, 70)
```

Comment:
hmm, so when we update the file format to a new version, then this test would start to raise an error because the function is trying to trim a v0.4.2 file when that's not the latest version anymore.  So maybe it'd be best to allow the test suite to not be brittle to have a kwarg boolean argument to the function, something like `allow_trimming_of_old_file_versions=False`, and then the tests could set it to true